draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -944,8 +946,25 @@ following conditions holds:
 
 ## "Same-site" and "cross-site" Requests  {#same-site-requests}
 
-A request is "same-site" if its target's URI's origin's registrable domain
-is an exact match for the request's client's "site for cookies", or if the
+Two origins, A and B, are considered same-site if the following algorithm returns true:
+1.  If A and B are both scheme/host/port triples then
+
+    1.  If A's scheme does not equal B's scheme, return false.
+
+    2.  Let hostA be A's host, and hostB be B's host.
+
+        1.  If hostA equals hostB and hostA's registrable domain is non-null, return true.
+
+        2.  If hostA's registrable domain equals hostB's registrable domain and is non-null, return true.
+
+2.  If A and B are both the same origin, return true.
```

Comment:
This might accidently return true for cases where we don't want it to
A= https://github.io
B= https://github.io

This should return false because the registrable domain of A & B is null, but with this algorithm we fall through and just check that they're the same which will return true.

Reply:
Fixed this.